packages/next/src/client/components/links.ts

```diff
@@ -364,7 +368,7 @@ function prefetchWithOldCacheImplementation(instance: PrefetchableInstance) {
     // note that `appRouter.prefetch()` is currently sync,
     // so we have to wrap this call in an async function to be able to catch() errors below.
     return instance.router.prefetch(instance.prefetchHref, {
-      kind: instance.kind,
+      kind: convertFetchStrategyToPrefetchKind(instance.fetchStrategy),
```

Comment:
Similarly, this is the only place this conversion ever happens. Should just inline it.